modules/network/main.tf

```diff
@@ -0,0 +1,177 @@
+  cidr_block           = var.base_cidr_block
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+
+  tags = {
+    Name = "${var.env_name}-VPC"
+  }
+}
+
+resource "aws_internet_gateway" "main" {
+  vpc_id = aws_vpc.main.id
+
+  tags = {
+    Name = "Main IGW"
+  }
+}
+
+resource "aws_subnet" "public_1" {
+  vpc_id            = aws_vpc.main.id
+  cidr_block        = cidrsubnet(aws_vpc.main.cidr_block, 8, 1)
+  availability_zone = "us-east-1a"
+
+  tags = {
+    Name = "Public Subnet 1"
+  }
+}
+
+resource "aws_subnet" "public_2" {
+  vpc_id            = aws_vpc.main.id
+  cidr_block        = cidrsubnet(aws_vpc.main.cidr_block, 8, 2)
+  availability_zone = "us-east-1b"
+
+  tags = {
+    Name = "Public Subnet 2"
+  }
+}
+
+resource "aws_subnet" "private_1" {
+  vpc_id            = aws_vpc.main.id
+  cidr_block        = cidrsubnet(aws_vpc.main.cidr_block, 8, 3)
+  availability_zone = "us-east-1a"
+
+  tags = {
+    Name = "Private Subnet 1"
+  }
+}
+
+resource "aws_subnet" "private_2" {
+  vpc_id            = aws_vpc.main.id
+  cidr_block        = cidrsubnet(aws_vpc.main.cidr_block, 8, 4)
+  availability_zone = "us-east-1b"
+
+  tags = {
+    Name = "Private Subnet 2"
+  }
+}
+
+resource "aws_route_table" "public" {
+  vpc_id = aws_vpc.main.id
+
+  tags = {
+    Name = "${var.env_name}-public-routing-table"
+  }
+}
+
+resource "aws_route" "public" {
+  route_table_id         = aws_route_table.public.id
+  destination_cidr_block = "0.0.0.0/0"
+  gateway_id             = aws_internet_gateway.main.id
+}
+
+resource "aws_main_route_table_association" "main" {
+  vpc_id         = aws_vpc.main.id
+  route_table_id = aws_route_table.public.id
+}
+
+resource "aws_route_table_association" "public_1" {
+  route_table_id = aws_route_table.public.id
+  subnet_id      = aws_subnet.public_1.id
+}
+
+resource "aws_route_table_association" "public_2" {
+  route_table_id = aws_route_table.public.id
+  subnet_id      = aws_subnet.public_2.id
+}
+
+resource "aws_security_group" "alb_main" {
+  name        = "alb_security_group"
+  description = "Security group for ALB"
+  vpc_id      = aws_vpc.main.id
+
+  tags = {
+    Name = "alb_security_group"
+  }
+}
+
+resource "aws_security_group_rule" "allow_incoming_http" {
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 80
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  ipv6_cidr_blocks  = ["::/0"]
+  security_group_id = aws_security_group.alb_main.id
+}
+
+resource "aws_security_group_rule" "allow_incoming_https" {
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 443
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  ipv6_cidr_blocks  = ["::/0"]
+  security_group_id = aws_security_group.alb_main.id
+}
+
+resource "aws_security_group_rule" "outgoing_to_anywhere" {
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  cidr_blocks       = ["0.0.0.0/0"]
```

Comment:
Technically from the ALB we only allow the app port, for example, Phoenix app is running at 4000, to the ECS container, we don't allow our ALB can outgoing to anywhere

Reply:
yeah I put it to all as wasn't 100% of container port / refactoring but will change this for now 👍 

And yeah we won't need any other port as we will access RDS and things through the container

Reply:
only 4000 for the ingress

The rule of thumb is

- Do not allow all ports to all IPs, we try to limit the access as low as possible 👌